src/evo/deterministicmns.cpp

```diff
@@ -223,10 +223,30 @@ std::vector<CDeterministicMNCPtr> CDeterministicMNList::GetProjectedMNPayees(int
     std::vector<CDeterministicMNCPtr> result;
     result.reserve(nCount);
 
+    auto remaining_hpmn_payments = 0;
+    ForEachMNShared(true, [&](const CDeterministicMNCPtr& dmn) {
+        if (dmn->pdmnState->nLastPaidHeight == nHeight) {
+            // We found the last MN Payee.
+            // If the last payee is a HPMN, we need to check its consecutive payments and pay him again if needed
+            if (dmn->nType == MnType::HighPerformance && dmn->pdmnState->nConsecutivePayments < dmn_types::HighPerformance.voting_weight) {
+                remaining_hpmn_payments = dmn_types::HighPerformance.voting_weight - dmn->pdmnState->nConsecutivePayments;
+                for ([[maybe_unused]] auto _ : irange::range(remaining_hpmn_payments)) {
+                    result.emplace_back(dmn);
+                }
+            }
+        }
+        return;
+    });
+
     ForEachMNShared(true, [&](const CDeterministicMNCPtr& dmn) {
         result.emplace_back(dmn);
+        if (dmn->nType == MnType::HighPerformance) {
+            result.emplace_back(dmn);
+            result.emplace_back(dmn);
+            result.emplace_back(dmn);
+        }
     });
-    std::sort(result.begin(), result.end(), [&](const CDeterministicMNCPtr& a, const CDeterministicMNCPtr& b) {
+    std::sort(result.begin() + remaining_hpmn_payments, result.end(), [&](const CDeterministicMNCPtr& a, const CDeterministicMNCPtr& b) {
```

Comment:
Good question... that doesn't make sense to me either..

Reply:
@UdjinM6 @PastaPastaPasta That's because the last (current) payee can be a HPMN not entirely paid yet (4 blocks).
So we add the remaining payments in line 234, and then we want to sort only the future payments.

Reply:
ah, so we skip results added by the first loop here. ok.